### PR TITLE
Initial pass at ME pipeline with Newspaper3k

### PR DESCRIPTION
**Pull request by caiismyname**
Summary: 
    - Created `MetadataExtractor` class that extracts ME given a url
    - Created `ArticleData` class to contain ME, with `toJson` method for portability
    - Created `Outlet` class to encapsulate outlet data (name, category, locale, media type)

Task:
    - Initial pipeline attempt

Testing:
    None

Questions:
    - What is return type of `pull_from_search/feeds`?
    - Can we use a shared article class instead of putting it in the collection subdir?

### CODE REVIEW

File: metadata-extraction/metadata-extraction.py [hunk 1]
```diff
@@ -1 +1,49 @@
-print("All data has been extracted")
+from newspaper import Article
+from urllib.parse import urlparse
+
+import shared-types
+
+class MetadataExtractor:
+    def __init__(self, url, article_data=None):
```
Comment (brgirardeau) [inline metadata-extraction/metadata-extraction.py]: Please add docstrings to all functions and classes.

File: shared-types.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        self.url = url
+        self.headline = headline
+        self.description = description
+        self.publication_date = publication_date
+        self.authors = authors
+        self.body_text = body_text
+        self.outlet = outlet
+
+    def toJson(self):
+        jsonOutput = {}
```
Comment (brgirardeau) [inline shared-types.py]: nit: snake case ?

File: shared-types.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        return json.dumps(jsonOutput)
+
+
+class Outlet:
+    def __init__(self, name=None, media_type=None, locale=None, category=None, url=None):
+        self.name = name
+        self.media_type = media_type
+        self.locale = locale
+        self.category = category
+        self.url = url # Base URL for the outlet, not for a particular article
```
Comment (jarrodddunne) [inline shared-types.py]: Rename 'url' to 'domain' for clarity?

File: metadata_extraction/metadata_extraction.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+from newspaper import Article
```
Comment (jarrodddunne) [inline metadata_extraction/metadata_extraction.py]: run.py still refers to this as 'metadata-extraction' (with a hyphen).

Comment (caiismyname) [reply metadata_extraction/metadata_extraction.py]: addressed in run.py

**Pull request merged**